main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -2328,6 +2330,13 @@ object Weeder {
       weedCharSequence(chars) map {
         string => Ast.Constant.Str(string)
       }
+
+    case ParsedAst.Literal.Regex(sp1, chars, sp2) =>
```

Comment:
We need to add a check to make sure that we don't see a regex on the left side of a `match` case. See `Weeder.visitPattern`.

Reply:
(Though of course in the future a pattern match using regexes would be very nice)

Reply:
I've added another case to the `case ParsedAst.Pattern.Lit` match following the example of `Null` .